platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/DefaultConfigurationCache.kt

```diff
@@ -327,7 +327,7 @@ class DefaultConfigurationCache internal constructor(
                     val description = formatBootstrapSummary(
                         "%s as configuration cache cannot be reused because %s.",
                         buildActionModelRequirements.actionDisplayName.capitalizedDisplayName,
-                        checkedFingerprint.reason
+                        checkedFingerprint.reason.render()
```

Comment:
❓ What's the value of explicitly calling `render` with no arguments here when it's the same as `toString`?

Reply:
I tried to catch all locations where we relied on `toString()`, explicitly or not, and replaced with explicit calls to `render()`. I left `toString()` calling `render()` just to avoid breaking some call site I could not trigger in my tests, but otherwise we could stop doing that, and have default a `toString()` implementations (such as inherited from `Object|Any.toString()`).

Reply:
> otherwise we could stop doing that,

What would be the benefit?

Reply:
I wanted to make sure existing usages worked as expected. Searching callers of `Property.toString()` does not help, as the IDE finds all callers of `Object.toString()`, and it still misses implicit calling - see [this comment](https://github.com/gradle/gradle/pull/29597#discussion_r1646737800).

Having a specific method that returns the string representation, and change all call sites relying on `toString()`. to instead call the new method solves that. And since we have that method, and want to encourage clients to continue to use that method, it would be best that `toString()` does not delegate to it.